### PR TITLE
0x53 MSTORE8

### PR DESCRIPTION
**Pull request by danilowhk**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- 0x53 MSTORE8 opcode
-
-

### CODE REVIEW

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm

**Comment by danilowhk** [comment]
For some reason, I am getting "make format" error even though on my pc its showing that it doesn't have any

**Comment by abdelhamidbakhta** [comment]
> For some reason, I am getting "make format" error even though on my pc its showing that it doesn't have any

Hmm, weird. Will check tomorrow

**Comment by danilowhk** [comment]
Ok! The last one that this happened, Lucas merged into another branch and then made the PR. But it did request for him to run "make format", could it be that my "make format" is formatting in a different way? As there are some changes in this PR that I didn't do. But overall it's a small thing hehe

File: src/kakarot/memory.cairo [hunk 1]
```diff
@@ -97,6 +97,59 @@ namespace Memory {
...
+        alloc_locals;
+        let (new_memory: felt*) = alloc();
+        if (self.bytes_len == 0) {
+            Helpers.fill_zeros(fill_with=offset, arr=new_memory);
+        }
+
+        let is_offset_greater_than_length = is_le_felt(self.bytes_len, offset);
+        local max_copy: felt;
+
+        assert [new_memory + offset] = element.low;
```
Comment (LucasLvy) [inline src/kakarot/memory.cairo:120]:
```suggestion
        assert [new_memory + offset] = element;
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: src/kakarot/instructions/memory_operations.cairo [hunk 2]
```diff
@@ -300,4 +301,45 @@ namespace MemoryOperations {
...
+        let stack = ctx.stack;
+
+        // Stack input:
+        // 0 - offset: memory offset of the work we save.
+        // 1 - value: value from which the last byte will be extracted and stored in memory.
+        let (stack, offset) = Stack.pop(stack);
+        let (stack, value) = Stack.pop(stack);
+        let (quotient, remainder) = uint256_unsigned_div_rem(value, Uint256(256,0));
+
+        let memory: model.Memory* = Memory.store8(self=ctx.memory, element=remainder, offset=offset.low);
```
[the remark this answers is not in the document]
Comment (danilowhk) [reply src/kakarot/instructions/memory_operations.cairo:334]: If we use just the value instead of remainder, i think it would filter the first byte

**Pull request closed without merging**